tests/cpp/conv_unittest.cc

```diff
@@ -0,0 +1,29 @@
+#include <conversation.h>
+#include <gtest/gtest.h>
+
+void _TestConversationJSONRoundTrip(std::string templ_name) {
```

Comment:
note for future, we can consider bind to python and do some testing over python later, no action needed now